generic/Storage.c

```diff
@@ -39,6 +39,26 @@ static int torch_Storage_(new)(lua_State *L)
       lua_pop(L, 1);
     }
   }
+  else if(lua_type(L, index) == LUA_TUSERDATA)
+  {
+    if (allocator)
+      THError("Passing allocator not supported when using storage views");
+
+    THStorage *src = luaT_checkudata(L, index, torch_Storage);
+    real *ptr = src->data;
```

Comment:
@dominikgrewe we removed the check here, it turns out that cascading views actually works fine, and has the behavior you would expect (cascaded offsets).